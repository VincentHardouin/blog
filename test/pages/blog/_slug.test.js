import { shallowMount } from '@vue/test-utils';
import BlogPost from '@/pages/blog/_slug';

describe('pages | blog | _slug', () => {
  it('should render blog post', () => {
    const wrapper = shallowMount(BlogPost, {
      data() {
        return {
          article: {
            title: 'Article title',
            description: 'description',
            seoImageUrl: 'https://example.net',
            content: 'Content',
            date: '2021-10-28',
          },
        };
      },
      computed: {
        date() {
          return 'formattedDate';
        },
        content() {
          return 'content';
        },
      },
    });

    expect(wrapper.find('.article__title').text()).toBe('Article title');
    expect(wrapper.find('.article__date').text()).toBe('formattedDate');
    expect(wrapper.find('.article__content').text()).toBe('content');
  });
});
