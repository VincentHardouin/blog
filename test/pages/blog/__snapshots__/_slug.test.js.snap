// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages | blog | _slug should render blog post 1`] = `
<div
  class="article"
>
  <div
    class="article__banner"
  />
   
  <div
    class="article__container"
  >
    <div
      class="article__date"
    >
      formattedDate
    </div>
     
    <h1
      class="article__title"
    >
      Article title
    </h1>
     
    <div
      class="article__content"
    >
      content
    </div>
  </div>
</div>
`;
